.github/workflows/release.yml

```diff
@@ -0,0 +1,116 @@
+  # constraints, which restricts the build plan. This mide hide build errors,
+  # because a more lenient build plan, e.g., a build plan including only the
+  # public libraries, could lead to compilation errors.
+  build-release:
+    name: Build for release
+    runs-on: ${{ matrix.os }}
+    strategy:
+       fail-fast: false
+       matrix:
+         ghc: ["8.10.7", "9.6.3"]
+         cabal: ["3.10.1.0"]
+         os: [ubuntu-latest]
+
+    steps:
+    - uses: actions/checkout@v4
+
+    - name: Install base libraries
+      uses: input-output-hk/actions/base@latest
+      with:
+        use-sodium-vrf: false
+
+    - name: Install Haskell
+      uses: input-output-hk/actions/haskell@latest
+      id: setup-haskell
+      with:
+        ghc-version: ${{ matrix.ghc }}
+        cabal-version: ${{ matrix.cabal }}
+
+    - name: Configure Cabal
+      run: |
+        cp .github/workflows/release/cabal.project .
+        rm -f cabal.project.freeze
+        rm -f cabal.project.local
+
+    - name: Update Cabal (Hackage and CHaP)
+      run: |
+        cabal clean
+        cabal update
+
+    # We create a `dependencies.txt` file that can be used to index the cabal
+    # store cache.
+    #
+    # We do not use `plan.json` directly because adding a dependency to our
+    # Cabal files which was already present somewhere else would result in a
+    # diferent plan, even though the set of dependencies is the same.
+    #
+    # In the future we should consider using `cabal-cache` like in the
+    # `cardano-node`'s GitHub workflow.
+    - name: Record dependencies to be used as cache keys
+      id: record-deps
+      run: |
+        cabal build all --dry-run --minimize-conflict-set
+        cat dist-newstyle/cache/plan.json \
+        | jq '.["install-plan"][].id' \
+        | sort \
+        | uniq > dependencies.txt
+
+    # This is used for caching the cabal store.
+    - name: Store month number as environment variable
+      run:  echo "MONTHNUM=$(/usr/bin/date -u '+%m')" >> $GITHUB_ENV
+
+    # Restore the cabal store cache.
+    #
+    # When we restore a previous cache and store a new key, we store both files
+    # that were already in the cache, and new files that were created. To
+    # prevent the cache from growing to quickly, we reset the cache each month.
+    - name: "Restore cache for cabal-release"
+      uses: actions/cache/restore@v3
+      id: restore-cabal-release
+      with:
+        path: ${{ steps.setup-haskell.outputs.cabal-store }}
+        key: cabal-release-${{ runner.os }}-${{ matrix.ghc }}-${{ env.MONTHNUM }}-${{ hashFiles('dependencies.txt') }}
+        restore-keys: |
+          cabal-release-${{ runner.os }}-${{ matrix.ghc }}-${{ env.MONTHNUM }}-${{ hashFiles('dependencies.txt') }}
+          cabal-release-${{ runner.os }}-${{ matrix.ghc }}-${{ env.MONTHNUM }}-
+
+    - name: Build dependencies
+      id: build-dependencies
+      run: cabal build --only-dependencies all -j
```

Comment:
Do we really need to build the project? can't a cabal-plan diff be enough?

Reply:
i.e. I think the libraries being `buildable` is much weaker than ensuring there are minimal changes wrt dependencies. In general I think we assume always that dependencies won't change a lot when testing and when not testing. But with this workflow they can change entirely as long as the project keeps building for the libs only.

Reply:
Good point -- that means it can also be a regular sanity check for PRs. I'll do that instead